constantine/constantine-jni/src/main/c/ethereum_evm_precompiles.c

```diff
@@ -0,0 +1,69 @@
+#include <jni.h>
+#include "org_hyperledger_besu_nativelib_constantine_LibConstantineEIP196.h"
+#include <constantine.h>
+#include <stdio.h>
+
+void printByteArray(const char* label, const byte* array, size_t len) {
+    printf("%s: [", label);
+    for (size_t i = 0; i < len; i++) {
+        printf("%02x", array[i]);
+        if (i < len - 1) {
+            printf(", ");
+        }
+    }
+    printf("]\n");
+}
+
+JNIEXPORT jint JNICALL Java_Constantine_ctt_1eth_1evm_1bn254_1g1add(JNIEnv *env, jobject obj, jbyteArray jr, jint r_len, jbyteArray jinputs, jint inputs_len) {
+    jbyte *r = (*env)->GetByteArrayElements(env, jr, NULL);
+    jbyte *inputs = (*env)->GetByteArrayElements(env, jinputs, NULL);
```

Comment:
Unfortunately I'm not familiar with Java at all but it seems like what is also being done in c-kzg-4844 so it's likely the expected way to wrap:

https://github.com/ethereum/c-kzg-4844/blob/1bccee0878ffc80efe8741afdb5793ef9105aa35/bindings/java/c_kzg_4844_jni.c#L148-L179